tools/deployment/presto-clp/scripts/generate-split-filter-file.py

```diff
@@ -0,0 +1,182 @@
+        return 1
+    if json_output_file.exists() and json_output_file.is_dir():
+        logger.error("Output path is a directory: %s", json_output_file)
+        return 1
+
+    datasets = _get_dataset_names(archives_dir)
+    if None == datasets:
+        logger.error("No datasets found in %s. Did you compress any logs yet?", archives_dir)
+        return 1
+
+    try:
+        timestamp_keys_by_dataset = _prompt_timestamp_keys(datasets)
+    except KeyboardInterrupt:
+        logger.error("Interrupted while collecting timestamp keys.")
+        return 1
+
+    split_filters = _construct_split_filters(datasets, timestamp_keys_by_dataset)
+    if split_filters is None:
+        logger.error("Missing timestamp key(s) for dataset(s).")
+        return 1
+
+    try:
+        with open(json_output_file, "w") as json_file:
+            json.dump(split_filters, json_file, indent=2)
+    except OSError as e:
+        logger.error("Failed to write output file %s: %s", json_output_file, e)
+        return 1
+
+    return 0
+
+
+def _get_dataset_names(archives_dir: Path) -> Optional[List[str]]:
+    """
+    Return the names of first-level subdirectories in <clp-package-dir>/var/data/archives. Each
+    subdirectory name is treated as a dataset name.
+
+    :param archives_dir:
+    :return: List of dataset names. None if there are no datasets compressed.
+    """
+
+    datasets = sorted([p.name for p in archives_dir.iterdir() if p.is_dir()])
+    return datasets if datasets else None
+
+
+def _prompt_timestamp_keys(datasets: List[str]) -> Dict[str, str]:
+    """
+    Prompt the user to provide the timestamp key for each dataset. If the user presses Enter, falls
+    back to `DEFAULT_TIMESTAMP_KEY`.
+
+    :param datasets:
+    :return: Dictionary mapping dataset names to their timestamp keys.
+    """
+    print(
+        "\nPlease enter the timestamp key that corresponds to each of your archived datasets."
+        "\nPress <Enter> to accept the default key.\n"
+    )
+
+    data_time_pairs: Dict[str, str] = {}
+    for dataset in datasets:
+        user_input = input(
+            f">>> {ANSI_BOLD}{dataset}{ANSI_RESET} [default key: {ANSI_BOLD}{DEFAULT_TIMESTAMP_KEY}{ANSI_RESET}]: "
+        )
+        key = DEFAULT_TIMESTAMP_KEY if 0 == len(user_input) else user_input
+        data_time_pairs[dataset] = key
+
+    return data_time_pairs
+
+
+def _construct_split_filters(
+    datasets: List[str],
+    timestamp_keys: Dict[str, str],
+) -> Optional[SplitFilterDict]:
+    """
+    Constructs a split filter for each dataset using a per-dataset timestamp key.
+
+    :param datasets:
+    :param timestamp_keys: Mapping from dataset name -> timestamp key.
+    :return: A SplitFilterDict containing all the SplitFilterRule objects for the JSON file.
+    :return: None if there are any datasets that don't have an associated timestamp key.
+    """
+
+    missing = [d for d in datasets if d not in timestamp_keys]
+    if 0 < len(missing):
```

Comment:
imo it's actuall very confusing to write if `0 < len(missing)`.
Two notes:
1. IIRC, We only need to put the constant on the write if it is a `==` and the rhs is a variable that you can assign value. in this case, you can write `if len(missing) > 0`.
2. I feel `if len(missing) != 0` is more straight forward.